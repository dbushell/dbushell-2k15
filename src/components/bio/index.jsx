import React from 'react';
import PropTypes from 'prop-types';

const Bio = props => {
  const attr = {
    className: 'b-bio',
    role: 'complementary',
    itemScope: true,
    itemType: 'http://schema.org/Person'
  };
  return (
    <section {...attr}>
      <div className="b-bio__image">
        <img src={props.imageSrc} srcSet={props.imageSrcset} alt={props.imageAlt}/>
      </div>
      <div className="b-bio__main">
        <h3 itemProp="name"><a href={props.href}>{props.title}</a></h3>
        <p itemProp="description">{props.text}</p>
      </div>
    </section>
  );
};

Bio.propTypes = {
  imageSrc: PropTypes.string,
  imageSrcset: PropTypes.string,
  imageAlt: PropTypes.string,
  title: PropTypes.string,
  href: PropTypes.string,
  text: PropTypes.string
};

Bio.defaultProps = require('./defaults');

export default Bio;
