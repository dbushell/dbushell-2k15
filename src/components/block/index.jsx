import React from 'react';
const Block = props => {
  if (props.main) {
    return (
      <main className={['c-main'].concat(props.classList).join(' ').trim()}>
        {props.children}
      </main>
    );
  }
  return (
    <div className={['b-block'].concat(props.classList).join(' ').trim()}>
      {props.children}
    </div>
  );
};
export default Block;
