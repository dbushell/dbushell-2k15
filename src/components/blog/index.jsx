import React, {PropTypes} from 'react';
import {Time} from '../';

const Blog = props => {
  const attr = {
    className: 'b-blog',
    role: 'navigation'
  };
  return (
    <aside {...attr}>
      <div className="b-blog__title">
        <h3>{props.heading}</h3>
      </div>
      <ul className="b-blog__list">
        {props.items.map(item => (
          <li className="b-blog__item" key={item.id}>
            <a rel="bookmark" href={item.href}>{item.title}</a>
            <Time date={item.date}/>
          </li>
        ))}
      </ul>
    </aside>
  );
};

Blog.propTypes = {
  heading: PropTypes.string,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      title: PropTypes.string,
      href: PropTypes.string,
      date: PropTypes.number
    })
  )
};

Blog.defaultProps = require('./defaults');

export default Blog;
