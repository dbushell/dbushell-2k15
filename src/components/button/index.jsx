import React from 'react';

/**
 * ButtonLabel
 */

export const ButtonLabel = props => {
  return (
    <span className="e-button__label">{props.children}</span>
  );
};

/**
 * Button
 */

const Button = props => {
  const attr = {
    className: 'e-button'
  };
  if (props.bg1) {
    attr.className += ' e-button--bg1';
  }
  if (props.bg2) {
    attr.className += ' e-button--bg2';
  }
  if (props.shadow) {
    attr.className += ' e-button--shadow';
  }
  const label = <ButtonLabel>{props.text}</ButtonLabel>;
  let button;
  if (props.submit) {
    button = <button type="submit" {...attr}>{label}</button>;
  } else {
    button = <a href={props.href} {...attr}>{label}</a>;
  }
  return button;
};

// Button.propTypes = {
//   text: PropTypes.string.isRequired,
//   href: PropTypes.string,
//   submit: PropTypes.bool,
//   bg1: PropTypes.bool,
//   bg2: PropTypes.bool,
//   shadow: PropTypes.bool
// };

Button.defaultProps = {
  text: 'Button'
};

export default Button;
