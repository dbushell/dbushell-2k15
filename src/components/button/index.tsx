import React from 'react';

interface ButtonLabelProps {
  text: string
}

const ButtonLabel: React.SFC<ButtonLabelProps> = props => {
  return (
    <span className="e-button__label">{props.text}</span>
  );
};

interface ButtonProps {
  text: string,
  href?: string,
  submit?: boolean,
  bg1?: boolean,
  bg2?: boolean,
  shadow?: boolean
}

const Button: React.SFC<ButtonProps> = props => {
  const attr = {
    className: 'e-button'
  };
  if (props.bg1) {
    attr.className += ' e-button--bg1';
  }
  if (props.bg2) {
    attr.className += ' e-button--bg2';
  }
  if (props.shadow) {
    attr.className += ' e-button--shadow';
  }
  const label = <ButtonLabel text={props.text}/>;
  let button;
  if (props.submit) {
    button = <button type="submit" {...attr}>{label}</button>;
  } else {
    button = <a href={props.href} {...attr}>{label}</a>;
  }
  return button;
};

Button.defaultProps = {
  text: 'Button'
};

export default Button;
