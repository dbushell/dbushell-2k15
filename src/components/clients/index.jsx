import React, {PropTypes} from 'react';
import {Block} from '../';
import Button from '../button';

const Clients = props => {
  return (
    <div className="c-clients u-dark">
      <Block>
        <div className="c-clients__header">
          <h2>{props.heading}</h2>
        </div>
        {props.blockquotes.map(item => (
          <blockquote key={item.id}>
            <p className="p--large p--quote">{item.quote}</p>
            <p className="p--small"><cite>{item.href ?
              <a href={item.href}>{item.cite}</a> : item.cite}</cite></p>
          </blockquote>
        ))}
        <div className="c-clients__footer">
          <Button {...props.button}/>
        </div>
      </Block>
    </div>
  );
};

Clients.propTypes = {
  heading: PropTypes.string,
  button: PropTypes.shape({...Button.propTypes}),
  blockquotes: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.any,
    quotation: PropTypes.string,
    cite: PropTypes.string,
    href: PropTypes.string
  }))
};

Clients.defaultProps = require('./defaults');

export default Clients;
