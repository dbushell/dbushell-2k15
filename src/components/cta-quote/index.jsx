import React from 'react';

const CTAQuote = () => {
  return (
    <div className="boxed reversed">
      <h3>Get a free quote&hellip;</h3>
      <p>Interested in working with me? <a href="/contact/">Send me an email</a> with your requirements and Iâ€™ll happily provide a free quote and let you know my availability.</p>
      <a href="/contact/" className="button">Get in Touch</a>
    </div>
  );
};

export default CTAQuote;

