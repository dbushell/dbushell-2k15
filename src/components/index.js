import Bio from './bio';
import Block from './block';
import Blog from './blog';
import Button from './button';
import Cta from './cta';
import Excerpt from './excerpt';
import Field from './field';
import Footer from './footer';
import Headline from './headline';
import Icon from './icon';
import Label from './label';
import Nav from './nav';
import Newsletter from './newsletter';
import Prose from './prose';
import Small from './small';
import Time from './time';

export {
  Bio,
  Block,
  Blog,
  Button,
  Cta,
  Excerpt,
  Field,
  Footer,
  Headline,
  Icon,
  Label,
  Nav,
  Newsletter,
  Prose,
  Small,
  Time
};
