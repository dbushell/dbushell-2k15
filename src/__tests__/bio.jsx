import React from 'react';
import {Bio} from '../components';
import {outerHTML} from './utils';

describe('Bio component', () => {
  it(`renders with defaults`, () => {
    expect(
      outerHTML(
        <Bio/>
      )
    ).toMatchSnapshot();
  });
});
