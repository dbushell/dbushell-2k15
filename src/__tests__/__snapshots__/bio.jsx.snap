// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio component renders with defaults 1`] = `
"<section class=\\"b-bio\\" role=\\"complementary\\" itemscope=\\"\\" itemtype=\\"http://schema.org/Person\\">
  <div class=\\"b-bio__image\\"><img src=\\"/assets/img/me3@1x.jpg\\" srcset=\\"/assets/img/me3@1x.jpg 1x, /assets/img/me3@2x.jpg 2x\\" alt=\\"David Bushell\\"></div>
  <div class=\\"b-bio__main\\">
    <h3 itemprop=\\"name\\"><a href=\\"/about/\\">David Bushell</a></h3>
    <p itemprop=\\"description\\">Based in the UK, near Manchester, I freelance for small businesses, start-ups, individuals, and fellow web agencies &#x2014; all over the world.</p>
  </div>
</section>"
`;
